cmake/Modules/FindMKL.cmake

```diff
@@ -122,10 +126,9 @@ set(INTEL_ROOT "/opt/intel" CACHE PATH "Folder contains intel libs")
     list(APPEND MKL_LIBRARIES ${${__mkl_lib_upper}_LIBRARY})
   endforeach()
 
-
   if(NOT MKL_USE_SINGLE_DYNAMIC_LIBRARY)
-    if (MKL_USE_STATIC_LIBS)
-      set(__iomp5_libs iomp5 libiomp5mt.lib)
+    if (MKL_USE_STATIC_LIBS AND UNIX)
+      set(__iomp5_libs libiomp5.a libiomp5mt.lib)
```

Comment:
Hmm, why we need static link libiomp5.a?

Reply:
Otherwise symbols are not resolved at link time and the llvm openmp will be used.

Reply:
Not in favor of tail wagging the dog solution, bending to allow mkl to behave foolishly.
Also, this doesn;t help transitive dependencies, i.e. some other clang library has pulled in libomp, and then you still have the problem.  I believe the fix is to squash mkl's bad behavior, possily through static linking or some other way.

Reply:
@cjolivier01 I'll remove testing MKL builds from this PR and you can contribute a separate PR that fixes MKL build and enables tests.

Reply:
Btw, if you look a few line above, you see that we are linking MKL statically.
Even if MKL is statically linked, dynamic linker may load llvm openmp as long as llvm openmp is pulled in by clang (depending on the link order)

Reply:
@leezu, I'm thinking about static linking MKL libraries and leaving omp dynamic link to mxnet.so. It should not have any issue about unresolved symbol. But please note that if linking libiomp5.so or llvm libomp.so, the MKL library should be `libmkl_intel_thread.a`, otherwise if libgomp.so, the MKL library should be `libmkl_gnu_thread.a`. Refer to the changes to makefile: https://github.com/apache/incubator-mxnet/pull/17693/files